Run `istio-cni` in `kube-system`

samcrang opened the pull request:
We are seeing this error when running the `istio-cni` `DaemonSet` in
`istio-system`:

```
Warning  FailedCreate  7m54s (x80 over 3h38m)  daemonset-controller
Error creating: pods "istio-cni-node-" is forbidden: pods with
system-node-critical priorityClass is not permitted in istio-system
namespace
```

samcrang committed:
Run `istio-cni` in `kube-system`

We are seeing this error when running the `istio-cni` `DaemonSet` in
`istio-system`:

```
Warning  FailedCreate  7m54s (x80 over 3h38m)  daemonset-controller
Error creating: pods "istio-cni-node-" is forbidden: pods with
system-node-critical priorityClass is not permitted in istio-system
namespace
```

This seems to have been caused after we did an upgrade of Istio (v1.1.4
=> v1.1.12)[1].

Documentation for upgrading Istio seems to suggest this is a required
step[2].

[1] https://github.com/alphagov/gsp/pull/350
[2] https://github.com/istio/istio.io/pull/4518/files#diff-34ca68b980d4edf68cb644bd257c549a

issyl0 reviewed (commented):
I'm not sure I understand why we're moving bits of Istio around out of the `istio-system` namespace? If it can't be updated where it is now (presumably that's not changed since we created it) then what's going on?

samcrang commented:
[We did an upgrade of Istio](https://github.com/alphagov/gsp/pull/350) and now things are broken :-( 

Possible related to [this](https://github.com/istio/istio/issues/14977). 🤷🏻‍♂️

issyl0 reviewed (dismissed):
That context helps a lot. Thanks!

samcrang commented:
I'll update the commit message to be more betterer.

issyl0 reviewed (dismissed):
:heavy_plus_sign: :100: for context.

samcrang committed:
Remove `istio-cni` from the `istio-system` namespace

In accordance with these upgrade steps[1].

[2] https://github.com/istio/istio.io/pull/4518/files#diff-34ca68b980d4edf68cb644bd257c549a

This commit should be reverted once the upgrade has been rolled out.

samcrang commented:
@adityapahuja @issyl0 I tweaked the commit messages a bit and added a revertable commit which cleans up (I hope) the currently installed `istio-cni`.

adityapahuja commented on pipelines/deployer/deployer.yaml:
```diff
@@ -217,6 +218,14 @@ apply_cluster_chart: &apply_cluster_chart
         --values config/${CONFIG_VALUES_PATH} \
         --set global.runningOnAws=true \
         gsp/gsp-istio-*.tgz
+      kubectl delete -R -f manifests/gsp-isti/istio-cni
```
Can we change `gsp-isti` to `gsp-istio`? Or is `gsp-isti` valid directory?

samcrang replied:
Oof. You're right. Good catch.
